shim/src/evaluators.js

```diff
@@ -0,0 +1,35 @@
+import { evaluate } from './evaluate';
+import { setPrototypeOf } from './commons';
+
+function getEvalEvaluator(sandbox) {
+  const o = {
+    // trick to set the name of the function to "eval"
+    eval: function(sourceText) {
+      // console.log(`Shim-Evaluation: "${sourceText}"`);
+      return evaluate(sourceText, sandbox);
+    }
+  };
+  setPrototypeOf(o.eval, sandbox.Function);
+  o.eval.toString = () => 'function eval() { [shim code] }';
+  return o.eval;
+}
+
+function getFunctionEvaluator(sandbox) {
+  const { confinedWindow } = sandbox;
```

Comment:
"window" is browser terminology, not host-independent js terminology. In any case, what's the difference between sandbox.unsafeGlobal and sandbox.confinedWindow ?

Reply:
Outdated code from the V1 of the shim.